produce_request.go

```diff
@@ -93,8 +93,12 @@ func (p *ProduceRequest) decode(pd packetDecoder) error {
 			if messageSetSize == 0 {
 				continue
 			}
+			msgSetDecoder, err := pd.getSubset(int(messageSetSize))
+			if err != nil {
+				return err
+			}
```

Comment:
This fixes decoding of produce requests with multiple message sets. This whole `decode` method was just added by @horkhe a few days ago, and is only used for tests so far, none of which caught this case until I wrote the one above.